plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/task/TaskManagerImpl.java

```diff
@@ -308,8 +312,17 @@ public Path getStatisticsFolder(@NotNull DBTTask task) {
     @Override
     public DBTTaskRunStatus runTask(@NotNull DBRProgressMonitor monitor, @NotNull DBTTask task, @NotNull DBTTaskExecutionListener listener) throws DBException {
         final TaskRunJob job = createJob((TaskImpl) task, listener);
-        final IStatus result = job.runDirectly(monitor);
-        final Throwable error = result.getException();
+        job.schedule();
+        if (serviceJob == null) {
+            serviceJob = new ServiceJob();
+            serviceJob.schedule();
+        }
+        try {
+            job.join();
```

Comment:
You changed core behavior. 
We mustn't schedule new job here. 
We mustn't call job.join() anywhere.